options.go

```diff
@@ -262,7 +264,51 @@ func AddrsFactory(factory config.AddrsFactory) Option {
 		if cfg.AddrsFactory != nil {
 			return fmt.Errorf("cannot specify multiple address factories")
 		}
-		cfg.AddrsFactory = factory
+		if factory == nil {
+			return fmt.Errorf("cannot specify a nil address factory")
+		}
+
+		var log = logging.Logger("addrs-factory")
+
+		cfg.AddrsFactory = func(in []ma.Multiaddr) []ma.Multiaddr {
+			// This wrapper does the following:
+			// - If the user returns a webtransport multiaddr that doesn't have
+			//   a certhash, and all our known addrs have a certhash, copy the
+			//   certhash over.
+			//   - If we only have a webtransport multiaddr without a certhash
+			//     (client passed a static tls config), don't copy.
```

Comment:
We don't support static TLS configs. As this is just a hack until we have a proper fix, we can reduce complexity here by hardcoding that.

Reply:
I'll warn instead since it's unexpected.

Reply:
log.error instead. We shouldn't hit that case, but if we do it's good to highlight it.